### PR TITLE
Localization system implementation

### PR DESCRIPTION
**Pull request by paolina92**
- Creation of a Context Class and add getBrowserLanguage method inside
- Creation of a Langstring Object to manage localization string
- Creation of the root reducer
- Synchronisation of of translations with store

### CODE REVIEW

**Comment by paolina92** [comment]
I added the frontend router on the same pull request / branch.

File: assembl/static2/tests/unit/containers/authentication.spec.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactTestRenderer from 'react-test-renderer';
+import Authentication from '../../../js/app/containers/authentication';
+
+describe('This test concern Authentication component', () => {
+  it('Should test React renderer', () => {
+    const renderer = ReactTestRenderer.create(
+      <Authentication />
+      );
+    const result = renderer.toJSON();
```
Comment (ayyazdaniaryan) [inline assembl/static2/tests/unit/containers/authentication.spec.js]: Hmm, ooh, so the `ReactTestRenderer` renders the entire virtual DOM and flushes the state onto it? So it's not actually binding the elements in the react DOM to the real DOM? So, what runs these tests? A node.js test runner? If so, then these are unit tests only, testing just the logic, and not the browser support and compatibility, right?

File: assembl/static2/tests/unit/utils/context.spec.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Context from '../../../js/app/utils/context';
+
+describe('This test concern Context Class', () => {
```
Comment (ayyazdaniaryan) [inline assembl/static2/tests/unit/utils/context.spec.js]: Me gusta

File: assembl/static2/js/app/index.jsx [hunk 3]
```diff
@@ -1,11 +1,27 @@
 import React from 'react';
```
Comment (ayyazdaniaryan) [inline assembl/static2/js/app/index.jsx]: Is there a reason why all the js files are now renamed to jsx?

File: assembl/static2/js/app/containers/authentication.jsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+
+const Authentication = () => (
+  <p><Translate value="authentication.panelTitle" /></p>
```
Comment (paolina92) [inline assembl/static2/js/app/containers/authentication.jsx]: Yes that's a failure of this system, that's why I suggest to use namespaces. I used this system in my last projects and it's not so bad. However, this system is simplest than Assembl v2 ;)

File: assembl/static2/js/app/routes.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import Authentication from './containers/authentication';
+import Home from './containers/home';
+import Ideas from './containers/ideas';
+import Synthesis from './containers/synthesis';
+import Debate from './containers/debate';
+
+export default (
+  <Route>
+    <Route path="/:slug/authentication" component={Authentication} />
+    <Route path="/:slug/home" component={Home} />
```
Comment (paolina92) [inline assembl/static2/js/app/routes.jsx]: As you wish but is it not most natural when the last parameter changes?

File: assembl/static2/js/app/utils/langString.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
+const LangString = {
```
Comment (paolina92) [inline assembl/static2/js/app/utils/langString.js]: I change for "Translations". Is it okay?

File: assembl/static2/tests/unit/containers/authentication.spec.js (hunk 1 above)
Comment (paolina92) [inline assembl/static2/tests/unit/containers/authentication.spec.js]: exact

File: assembl/static2/js/app/index.jsx (hunk 3 above)
Comment (paolina92) [inline assembl/static2/js/app/index.jsx]: Yes, it's better to use jsx extension when we have to use a react component. If not, we have to use js extension. Right?

File: assembl/static2/js/app/utils/context.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import LangString from './langString';
+
+class Context {
```
Comment (paolina92) [inline assembl/static2/js/app/utils/context.js]: Hmmm. Good question. I'm not sure that "context" is the good name. I need to have some global functions in the project. So I renamed it to "globalFunctions.js"? Is it okay? I need your feedback please.
However, the locale is already in the store with all translations.
The store.getState() method return this:
i18n{
    locale:"fr",
    translations:{
        de:{...},
        en:{...},
        fr:{...}
    }
}

File: assembl/static2/tests/unit/containers/authentication.spec.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactTestRenderer from 'react-test-renderer';
+import Authentication from '../../../js/app/containers/authentication';
```
Comment (paolina92) [inline assembl/static2/tests/unit/containers/authentication.spec.js]: I tried to use "~" but I broke my teeth... I need help to configure it

**Pull request merged**